lib/src/cpp/domain_participant/builtin/discovery_config/DiscoveryConfig.cpp

```diff
@@ -0,0 +1,531 @@
+        const std::string& profile_id,
+        const std::string& ignore_participant_flags)
+{
+    throw Unsupported("Unsupported");
+}
+
+void set_edp(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& edp)
+{
+    throw Unsupported("Unsupported");
+}
+
+void set_simple_edp_pubwriter_subreader(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& simple_edp_pubwriter_subreader)
+{
+    throw Unsupported("Unsupported");
+}
+
+void set_simple_edp_pubreader_subwriter(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& simple_edp_pubreader_subwriter)
+{
+    throw Unsupported("Unsupported");
+}
+
+void set_lease_duration(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& duration_sec,
+        const std::string& duration_nanosec)
+{
+    throw Unsupported("Unsupported");
+}
+
+void set_lease_duration_sec(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& duration_sec)
+{
+    throw Unsupported("Unsupported");
+}
+
+void set_lease_duration_nanosec(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& duration_nanosec)
+{
+    throw Unsupported("Unsupported");
+}
+
+void set_lease_announcement(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& announcement_sec,
+        const std::string& announcement_nanosec)
+{
+    throw Unsupported("Unsupported");
+}
+
+void set_lease_announcement_sec(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& announcement_sec)
+{
+    throw Unsupported("Unsupported");
+}
+
+void set_lease_announcement_nanosec(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& announcement_nanosec)
+{
+    throw Unsupported("Unsupported");
+}
+
+void set_initial_announcements_count(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& count)
+{
+    throw Unsupported("Unsupported");
+}
+
+void set_initial_announcements_period(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& period_sec,
+        const std::string& period_nanosec)
+{
+    throw Unsupported("Unsupported");
+}
+
+void set_initial_announcements_period_sec(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& period_sec)
+{
+    throw Unsupported("Unsupported");
+}
+
+void set_initial_announcements_period_nanosec(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& period_nanosec)
+{
+    throw Unsupported("Unsupported");
+}
+
+void set_client_announcement_period(
```

Comment:
This `set` function with both `sec` and `nanosec` arguments is no longer required, we changed it into two individual set functions for `sec` and `nanosec`